src/DurableTask.AzureStorage/AzureStorageOrchestrationService.cs

```diff
@@ -526,14 +580,30 @@ internal Task<IEnumerable<BlobLease>> ListBlobLeasesAsync()
 
             string taskHub = azureStorageClient.Settings.TaskHubName;
 
-            BlobLeaseManager inactiveLeaseManager = GetBlobLeaseManager(azureStorageClient, "inactive");
+            //Need to check for leases in Azure Table Storage. Scale Controller calls into this method.
+            
+            int partitionCount;
+            Table partitionTable = azureStorageClient.GetTableReference(azureStorageClient.Settings.PartitionTableName);
+            // Check if table partition manager is used. If so, get partition count from table.
+            // Else, get the partition count from the blobs.
+
+            if (await partitionTable.ExistsAsync())
```

Comment:
nit: let's place comments immediately above the lines they reference. No need to separate them with a newline

```suggestion
            //Need to check for leases in Azure Table Storage. Scale Controller calls into this method.
            int partitionCount;
            Table partitionTable = azureStorageClient.GetTableReference(azureStorageClient.Settings.PartitionTableName);

            // Check if table partition manager is used. If so, get partition count from table.
            // Else, get the partition count from the blobs.
            if (await partitionTable.ExistsAsync())
```